src/parcel_snoop_inference.jl

```diff
@@ -260,26 +260,103 @@ end
 
 ## parcel and supporting infrastructure
 
-function isprecompilable(mi::MethodInstance; excluded_modules=Set([Main::Module]))
+"""
+    isprecompilable(mod::Module, mi::MethodInstance)
+    isprecompilable(mi::MethodInstance; excluded_modules=Set([Main::Module]))
+
+Determine whether `mi` is able to be precompiled within `mod`. This requires
+that all the types in `mi`'s specialization signature are "known" to `mod`. See
+[`SnoopCompile.known_type`](@ref) for more information.
+
+`isprecompilable(mi)` sets `mod` to the module in which the corresponding method
+was defined. If `mod ∈ excluded_modules`, then `isprecompilable` returns `false`.
+
+If `mi` has been compiled by the time its defining module "closes" (the final
+`end` of the module definition) and `isprecompilable(mi)` returns `true`, then
```

Comment:
> If `mi` has been compiled by the time its defining module "closes" (the final
`end` of the module definition)

When might this happen? Would it mean that the defining module would need to call `mi`?

Reply:
If `mi` corresponds to `f(x)`, then just calling `f(x)` before the module closes will do it. For the inferrable callees of this method, it won't matter if it's inside or outside a `@compile_workload` block. For any runtime-dispatched ones, it will matter a lot. But if you're thinking about a single "flame" on the flamegraph, that's irrelevant, because one flame = the inferrable callees of the entrypoint (base of the flame).